### PR TITLE
Firefly-1338: Fixed: dce search race condition when launched via API

### PR DESCRIPTION
**Pull request by robyww**
#### Testing
- https://firefly-1338-dce-race.irsakudev.ipac.caltech.edu/
- Do a radar search and then, send to DCE

### CODE REVIEW

**Comment by lrebull** [comment]
Ooh !!! yes, this works!!!

File: src/firefly/js/ui/dynamic/DLGeneratedDropDown.js [hunk 1]
```diff
@@ -489,20 +494,33 @@ function DLGeneratedTableSearch({currentTblId, initArgs, sideBar, regHasUrl, url
...
-        if (initArgs?.urlApi?.execute && clickFuncRef.clickFunc && matchUrl) {
+        console.log(`in effect: regLoaded:${regLoaded}, regHasUrl:${regHasUrl}, matchUrl:${matchUrl}, clickFunc:${Boolean(onClick)}`);
+        if (initArgs?.urlApi?.execute && onClick && matchUrl) {
+            console.log('trying: execute once');
             executeInitOnce(true, () => {
+                console.log(`execute (${initArgs.urlApi.callId}) url: ${initArgs?.urlApi.url}`, initArgs?.urlApi);
                 setCallId(initArgs.urlApi.callId ?? 'none'); //forces one more render after unmount
                 dispatchMountFieldGroup(GROUP_KEY, false, false); // unmount to force to forget default so it will reinit
-                setTimeout(() => clickFuncRef?.clickFunc?.(),10);
+                setTimeout(() => onClick?.(),10);
```
Comment (dencoded) [inline src/firefly/js/ui/dynamic/DLGeneratedDropDown.js:520]: we don't have to do chaining `onClick?` here as we already tested it in if-condition above

**Review by loitly — APPROVED** [review]
I see why `useState` would help in this situation.

**Comment by lrebull** [comment]
it is working for me!!

**Review by dencoded — APPROVED** [review]
I understand the nature of the problem and solution works.

File: src/firefly/js/ui/dynamic/DLGeneratedDropDown.js (hunk 1 above)
Comment (robyww) [reply src/firefly/js/ui/dynamic/DLGeneratedDropDown.js]: thanks. fixed.

**Pull request merged**